index.bs

```diff
@@ -74,6 +75,125 @@ A user agent must not <a spec=HTML>delay the load event</a> of the <{link}> elem
 
 <p class="issue">TODO: We should define a list of allowable referrer policies to prerender with, and early-return if the above referrer policy is not in this list.
 
+<h2 id="speculation-rules">Speculation rules</h2>
+
+<h3 id="speculation-rules-dfns">Definitions</h3>
+
+A <dfn>speculation rule</dfn> is a [=struct=] with the following [=struct/items=]:
+* <dfn for="speculation rule">URLs</dfn>, an [=ordered set=] of [=URLs=]
+* <dfn for="speculation rule">requirements</dfn>, an [=ordered set=] of [=strings=]
+
+The only valid string for [=speculation rule/requirements=] to contain is "`anonymous-client-ip`".
+
+A <dfn>speculation rule set</dfn> is a [=struct=] with the following [=struct/items=]:
+* <dfn for="speculation rule set">prefetch rules</dfn>, a [=list=] of [=speculation rules=]
+
+<h3 id="speculation-rules-script">The <{script}> element</h3>
+
+<em>Note</em>: This section contains modifications to the corresponding section of [[HTML]].
+
+<p class="issue">We should consider whether we also want to make this execute even if scripting is disabled.</p>
```

Comment:
nit: Feel free to omit the end tag as is consistent with the other notes in this spec

Reply:
done